lxd/storage/drivers/driver_powerflex.go

```diff
@@ -0,0 +1,208 @@
+		d.config["powerflex.user.name"] = powerFlexDefaultUser
+	}
+
+	if d.config["powerflex.mode"] == "" {
+		if d.canSDC() {
+			d.config["powerflex.mode"] = "sdc"
+		}
+
+		// Prefer the NVMe/TCP mode.
+		if d.canNVMe() {
+			d.config["powerflex.mode"] = "nvme"
+		}
+	}
+
+	// Fail if no PowerFlex mode could be discovered.
+	if d.config["powerflex.mode"] == "" {
+		return fmt.Errorf("Failed to discover PowerFlex mode")
+	}
+
+	// PowerFlex volumes have to be at least 8GiB in size.
+	if d.config["volume.size"] == "" {
+		d.config["volume.size"] = powerFlexDefaultSize
+	}
+
+	return nil
+}
+
+// Create is called during pool creation and is effectively using an empty driver struct.
+// WARNING: The Create() function cannot rely on any of the struct attributes being set.
+func (d *powerflex) Create() error {
+	err := d.FillConfig()
+	if err != nil {
+		return err
+	}
+
+	// Discover one of the storage pools SDS services.
+	if d.config["powerflex.mode"] == "nvme" {
+		if d.config["powerflex.sds"] == "" {
+			pool, err := d.powerFlex().getStoragePool(d.config["powerflex.pool"])
+			if err != nil {
+				return err
+			}
+
+			// TODO: check status of SDS or if they are applicable (ipList.role, sdsState, membershipState, mdmConnectionState)
+			sds, err := d.powerFlex().getProtectionDomainSdsRelations(pool.ProtectionDomainID)
+			if err != nil {
+				return err
+			}
+
+			if len(sds) == 0 {
+				return fmt.Errorf("Failed to retrieve at least one SDS for the given storage pool: %q", pool.ID)
+			}
+
+			if len(sds[0].IPList) == 0 {
+				return fmt.Errorf("Failed to retrieve IP from SDS: %q", sds[0].Name)
+			}
+
+			d.config["powerflex.sds"] = sds[0].IPList[0].IP
+		}
+	}
+
+	if d.config["powerflex.mode"] == "sdc" {
+		return fmt.Errorf("not supported")
+	}
+
+	return nil
+}
+
+// Delete removes the storage pool from the storage device.
+func (d *powerflex) Delete(op *operations.Operation) error {
+	// Disconnect from the NVMe/TCP subsystem.
+	if d.config["powerflex.mode"] == "nvme" {
+		err := d.disconnectNVMeSubsys()
+		if err != nil {
+			return err
+		}
+	}
+
+	// If the user completely destroyed it, call it done.
+	if !shared.PathExists(GetPoolMountPath(d.name)) {
+		return nil
+	}
+
+	// On delete, wipe everything in the directory.
+	return wipeDirectory(GetPoolMountPath(d.name))
+}
+
+// Validate checks that all provide keys are supported and that no conflicting or missing configuration is present.
+func (d *powerflex) Validate(config map[string]string) error {
+	rules := map[string]func(value string) error{
+		"powerflex.user.name":      validate.IsAny,
+		"powerflex.user.password":  validate.IsAny,
+		"powerflex.gateway":        validate.IsRequestURL,
+		"powerflex.gateway.verify": validate.Optional(validate.IsBool),
+		"powerflex.pool":           validate.IsNotEmpty,
+		"powerflex.domain":         validate.Optional(validate.IsAny),
+		"powerflex.mode":           validate.Optional(validate.IsOneOf("nvme", "sdc")),
```

Comment:
Can we lose mentions of sdc until it works, that way we can introduce sdc support as a separate PR

Reply:
Removed the `sdc` option.